src/domain/kaivuilmoitusMuutosilmoitus/Attachments.tsx

```diff
@@ -0,0 +1,222 @@
+import { Trans, useTranslation } from 'react-i18next';
+import { Link, Notification } from 'hds-react';
+import { Box, Flex } from '@chakra-ui/react';
+import { useQueryClient } from 'react-query';
+import { useFormContext } from 'react-hook-form';
+import FileUpload from '../../common/components/fileUpload/FileUpload';
+import {
+  deleteAttachment,
+  getAttachmentFile,
+  uploadAttachment,
+} from '../application/muutosilmoitus/muutosilmoitusAttachmentsApi';
+import { getAttachmentFile as getApplicationAttachmentFile } from '../application/attachments';
+import FileDownloadList from '../../common/components/fileDownloadList/FileDownloadList';
+import {
+  FormSummarySection,
+  SectionItemContent,
+  SectionItemTitle,
+} from '../forms/components/FormSummarySection';
+import { ApplicationAttachmentMetadata } from '../application/types/application';
+import { MuutosilmoitusAttachmentMetadata } from '../application/muutosilmoitus/types';
+import { KaivuilmoitusMuutosilmoitusFormValues } from './types';
+import TextArea from '../../common/components/textArea/TextArea';
+import { AttachmentMetadata } from '../../common/types/attachment';
+
+type Props = {
+  applicationId: number;
+  muutosilmoitusAttachments: MuutosilmoitusAttachmentMetadata[];
+  originalAttachments?: ApplicationAttachmentMetadata[];
+};
+
+export default function Attachments({
```

Comment:
I was thinking if it would be possible to create a common component for here and kaivuilmoitus täydennys attachments as there is quite a lot of duplication